lib/ansible/modules/privileged_account_security/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,292 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: cyberark_authentication
+short_description: "Module for CyberArk Vault Authentication using Privileged Account Security Web Services SDK"
+author: "Edward Nunez (@enunez-cyberark)"
+version_added: "2.3"
+description:
+    - "Authenticates to CyberArk Vault using Privileged Account Security Web Services SDK and
+       creates a session fact that can be used by other modules. It returns an Ansible fact
+       called I(cyberarkSession). Every module can use this fact as C(cyberarkSession) parameter."
+
+
+options:
+    state:
+        default: present
+        choices: ['present', 'absent']
+        description:
+            - Specifies if an authentication logon/logoff and a cyberarkSession should be added/removed
+    userName:
+        required: False
+        description:
+            - The name of the user who will logon to the Vault
+    password:
+        required: False
+        description:
+            - The password of the user
+    newPassword:
+        required: False
+        description:
+            - The new password of the user. This parameter is optional, and enables you to change a password
+    apiBaseUrl:
+        required: False
+        description:
+            - A string containing the base URL of the server hosting CyberArk's Privileged Account Security Web Services SDK
+    validateCerts:
+        required: False
+        default: false
+        description:
+            - If C(false), SSL certificates will not be validated.  This should only
+              set to C(false) used on personally controlled sites using self-signed
+              certificates.
+    useSharedLogonAuthentication:
+        required: False
+        default: false
+        description:
+            - Whether or not Shared Logon Authentication will be used.
+    useRadiusAuthentication:
+        required: False
+        default: false
+        description:
+            - Whether or not users will be authenticated via a RADIUS server. Valid values are true/false
+    cyberarkSession:
+        required: False
+        description:
+            - Dictionary set by a CyberArk authentication containing the different values to perform actions on a logged-on CyberArk session
+'''
+
+EXAMPLES = '''
```

Comment:
in the case of the module cyberark_authentication... cyberarkSession is returned as a fact ... and you can use it as it in the example for the logoff where cyberarkSession is "{{ cyberarkSession }}"